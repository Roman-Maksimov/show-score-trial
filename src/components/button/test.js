import React from 'react';
import Button from './';
import renderer from 'react-test-renderer';

test('should render common button', () => {
  const component = renderer.create(
    <Button>OK</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render accent button', () => {
  const component = renderer.create(
    <Button accent>OK</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render custom button', () => {
  const component = renderer.create(
    <Button className="customButton">OK</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render custom accent button', () => {
  const component = renderer.create(
    <Button className="customButton" accent>OK</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
