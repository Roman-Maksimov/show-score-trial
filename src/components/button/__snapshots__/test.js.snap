// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render accent button 1`] = `
<button
  className="buttonAccent"
>
  <div
    className="inner"
  >
    OK
  </div>
</button>
`;

exports[`should render common button 1`] = `
<button
  className="button"
>
  <div
    className="inner"
  >
    OK
  </div>
</button>
`;

exports[`should render custom accent button 1`] = `
<button
  className="buttonAccent customButton"
>
  <div
    className="inner"
  >
    OK
  </div>
</button>
`;

exports[`should render custom button 1`] = `
<button
  className="button customButton"
>
  <div
    className="inner"
  >
    OK
  </div>
</button>
`;
